src/sentry/profiles/flamegraph.py

```diff
@@ -506,3 +510,65 @@ def get_profile_candidates_from_profiles(self) -> ProfileCandidates:
             "transaction": transaction_profile_candidates,
             "continuous": continuous_profile_candidates,
         }
+
+    def get_profile_candidates_from_spans(self) -> ProfileCandidates:
+        max_profiles = options.get("profiling.flamegraph.profile-set.size")
+        results = self.get_spans_based_candidates(query=self.query, limit=max_profiles)
+        transaction_profile_candidates: list[TransactionProfileCandidate] = [
+            {
+                "project_id": row["project.id"],
+                "profile_id": row["profile.id"],
+            }
+            for row in results["data"]
+            if row["profile.id"] is not None and row["profile.id"] != ""
+        ]
+
+        max_continuous_profile_candidates = max(
+            max_profiles - len(transaction_profile_candidates), 0
+        )
+        return {
+            "transaction": transaction_profile_candidates,
+            "continuous": self.get_chunks_for_profilers(
+                [
+                    ProfilerMeta(
+                        project_id=row["project.id"],
+                        profiler_id=row["profiler.id"],
+                        thread_id=row["thread.id"],
+                        start=row["precise.start_ts"],
+                        end=row["precise.finish_ts"],
+                    )
+                    for row in results["data"]
+                    if row["profiler.id"] is not None and row["thread.id"]
+                ],
+                max_continuous_profile_candidates,
+            ),
+        }
+
+    def get_spans_based_candidates(self, query: str | None, limit: int) -> EAPResponse:
+        # add constraints in order to fetch only spans with profiles
+        profiling_constraint = "(has:profile.id) or (has:profiler.id has:thread.id)"
+        if query is not None and len(query) > 0:
+            query += f" and {profiling_constraint}"
```

Comment:
Nit: To avoid creating a multiple intermediate strings

```suggestion
            query = f"{query} and {profiling_constraint}"
```